modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Iri.scala

```diff
@@ -84,12 +89,82 @@ object Iri {
   final def absolute(string: String): Either[String, AbsoluteIri] =
     urn(string) orElse url(string)
 
+  /**
+    * Attempt to construct a new RelativeIri from the argument as per the RFC 3987.
+    *
+    * @param string the string to parse as a relative iri
+    * @return Right(RelativeIri) if the string conforms to specification, Left(error) otherwise
+    */
+  final def relative(string: String): Either[String, RelativeIri] =
```

Comment:
we can now have a `final def apply(string: String): Either[String, Iri]` function in the `Iri` companion.